### ISSUE TITLE
model.reset_states() gets "NoneType" error when it contains a stateful LSTM and initial_state specified

### ISSUE THREAD

**Issue by rfernand2** [keras-team/keras#9417]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by rfernand2** [comment]
Here is a test case that generate the error.  Does anyone have an idea for a workaround to this bug?  Thanks.

# resetStates.py - simple test case for calling reset_states() on stateful LSTM model. #
from keras.models import Model
from keras.layers import Input, LSTM, Dense
from keras.layers import Dropout, BatchNormalization

latent_dim = 16
vocab_dim = 20
max_seq = 8

# sinple decoder model #
input_teacher = Input(batch_shape = (1, max_seq, vocab_dim))
input_left = Input(batch_shape = (1, latent_dim))
input_right = Input(batch_shape = (1, latent_dim))

#lstm = LSTM(latent_dim, stateful=True)(input_teacher)  # this line runs OK
lstm = LSTM(latent_dim, stateful=True)(input_teacher, initial_state=[input_left, input_right])   # this line gets NoneType error #
dense = Dense(vocab_dim)(lstm)

model = Model([input_teacher, input_left, input_right], [dense])
model.reset_states()
print("model.reset_states() succeeded")
